### PR TITLE
fix pkg-config handling to make cross-compiling work

### PR DESCRIPTION
**Pull request by vapier**
This series cleans up the pkg-config logic so that cross-compile works properly.  There's still some weirdness with the use of freetype-config/libpng-config, but I've ignored that as it only impacts the display and not the actual config/code generation.

### CODE REVIEW

**Comment by tacaswell** [comment]
This breaks every travis build because it can not find freetype (although it works in a conda env on arch both with and without their freetype package).  I think travis is a pretty 'standard' ubuntu set up so if it does not compile there it probably won't compile on most user systems.

@cgohlke @jbmohler How badly does this break windows?

@jenshnielsen @efiring Does this work on mac?

This PR may require (significant?) documentation on how to 'properly' set up compilers on more-or-less every system.

**Comment by mdboom** [comment]
It looks like this removes the fallbacks for when pkg-config isn't found.  We definitely need that, because we can't depend on pkg-config being there -- we only want to use it (and give it preference) if it's there and working.

**Comment by vapier** [comment]
i didn't touch the pkg-config fallbacks.  the 3rd patch dropped the -I/-L system paths which were always applied.  i guess the travis system is broken, but whatever -- i've reworked the 3rd patch so it adds those hardcodes only when pkg-config is not available.

**Comment by tacaswell** [comment]
I actually don't know if pkg-config is installed on travis or not, that is probably something we should check and make sure it is _not_ installed as that is the profile that is going to provide a bigger pain point for us.

**Comment by jklymak** [comment]
Closing as unlikely to be still relevant.  Feel free to re-open if I'm mistaken...

**Comment by vapier** [comment]
commit 64d8ac677a1cfe6c23372a8664a9c39b467dd5bc was merged later which was basically the same as my first commit (although mine used fewer loc).  *shrug*

the other two patches are still relevant.  i've rebased them now.

**Comment by vapier** [comment]
looks like i don't have the ability to re-open.  maybe only project members can do that.

**Comment by jklymak** [comment]
I can't re-open either.  I suspect it won't let you re-open if it conflicts? But a new PR against master would be easiest.  Sorry to create an inconvenience; I am just trying to weed through old PRs...

**Comment by vapier** [comment]
np.  i've moved this to #11218 now.

File: setupext.py [hunk 1]
```diff
@@ -252,29 +253,15 @@ def __init__(self):
...
+            self.pkgconfig = os.environ.get('PKG_CONFIG', 'pkg-config')
+            status, output = getstatusoutput("%s --help" % self.pkgconfig)
             self.has_pkgconfig = (status == 0)
             if not self.has_pkgconfig:
                 print("IMPORTANT WARNING:")
                 print(
                     "    pkg-config is not installed.\n"
                     "    matplotlib may not be able to find some of its dependencies")
 
-    def set_pkgconfig_path(self):
```
Comment (anntzer) [inline setupext.py]: Why was this removed?
Note that (unchecked...) this *may* be needed to make things work in a conda env which does install the libraries in the environment but where pkg-config is "outside" the env (because sysconfig.get_config_var("LIBDIR") will correctly point inside the env in that case).

**Comment by tacaswell** [comment]
If you have force-pushed to a branch you can not re-open.
